current usage against the org quota

AmitRoushan opened the pull request:
Provide current status of routes, service instances and memory usage of organization quota.
[#91309412](https://www.pivotaltracker.com/story/show/91309412)

If this implementation looks good then I would like to do PR for space quota [#91309436](https://www.pivotaltracker.com/story/show/91309436) also.

jpalermo commented on lib/cloud_controller/dependency_locator.rb:
```diff
@@ -201,6 +201,10 @@ def paginated_collection_renderer
       create_paginated_collection_renderer
     end
 
+    def object_collection_renderer
```
Has this method been added for a reason? We're only seeing it used in tests.

goehmen replied:
Love this.  Can't wait to expose it in the CLI.  I was going to try to
accomplish this via a plugin but happy to just hit CC endpoints!

Greg Oehmen
Cloud Foundry Product Manager
415.205.6596


On Thu, May 14, 2015 at 4:43 PM, Joseph Palermo <notifications@github.com>
wrote:

> In lib/cloud_controller/dependency_locator.rb
> <https://github.com/cloudfoundry/cloud_controller_ng/pull/378#discussion_r30374659>
> :
>
> > @@ -201,6 +201,10 @@ def paginated_collection_renderer
> >        create_paginated_collection_renderer
> >      end
> >
> > +    def object_collection_renderer
>
> Has this method been added for a reason? We're only seeing it used in
> tests.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cloudfoundry/cloud_controller_ng/pull/378/files#r30374659>
> .
>

AmitRoushan commented:
Hi @jpalermo 
Thanks for review.
We have added object_collection_renderer to test create_object_collection_renderer function only.

Thanks 
@AmitRoushan

jpalermo commented:
I'm not sure create_object_collection_renderer makes sense. A collection renderer is responsible for taking a collection of things and rendering them all, here you only have a single thing.

Similarly the `ObjectRenderer` should not have a `collection_transformer`, but just a `transformer`since it only deals with single objects and not collections.

AmitRoushan commented:
Hi @jpalermo 
Thanks again for reviewing and providing your valuable comments.
As per your suggestion, I have removed 'collection' from possible places.
Please check if modifications are inline with the expectations.

Thanks
@AmitRoushan

jpalermo commented on lib/cloud_controller/dependency_locator.rb:
```diff
@@ -201,6 +201,10 @@ def paginated_collection_renderer
       create_paginated_collection_renderer
     end
 
+    def created_object_renderer
```
This method is never used by quota_usage_populating_renderer, or any other production code. It should probably be removed.

Tests for the values passed to ObjectRenderer.new, should just be moved to the test for quota_usage_populating_renderer.

Since the implementation of quota_usage_populating_renderer could change, having a test on the created_object_renderer doesn't provide much use.

AmitRoushan commented:
Hi @jpalermo 
As per your suggestion, I have removed 'created_object_renderer' and added testcases for create_object_renderer in quota_usage_populating_renderer context.
I am not sure about why CI is not triggered.
Thanks again for review comments.

Thanks
@AmitRoushan
